pkg/templates/nonisolatedpod/template.go

```diff
@@ -0,0 +1,66 @@
+package nonisolatedpod
+
+import (
+	"fmt"
+
+	"golang.stackrox.io/kube-linter/pkg/check"
+	"golang.stackrox.io/kube-linter/pkg/config"
+	"golang.stackrox.io/kube-linter/pkg/diagnostic"
+	"golang.stackrox.io/kube-linter/pkg/extract"
+	"golang.stackrox.io/kube-linter/pkg/lintcontext"
+	"golang.stackrox.io/kube-linter/pkg/objectkinds"
+	"golang.stackrox.io/kube-linter/pkg/templates"
+	"golang.stackrox.io/kube-linter/pkg/templates/nonisolatedpod/internal/params"
+	networkingV1 "k8s.io/api/networking/v1"
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+const (
+	templateKey = "non-isolated-pod"
+)
+
+func init() {
+	templates.Register(check.Template{
+		HumanName:   "Non Isolated Pods",
+		Key:         templateKey,
+		Description: "Flag Pod that is not selected by any networkPolicy",
+		SupportedObjectKinds: config.ObjectKindsDesc{
+			ObjectKinds: []string{objectkinds.NetworkPolicy},
+		},
+		Parameters:             params.ParamDescs,
+		ParseAndValidateParams: params.ParseAndValidate,
+		Instantiate: params.WrapInstantiateFunc(func(_ params.Params) (check.Func, error) {
+			return func(lintCtx lintcontext.LintContext, object lintcontext.Object) []diagnostic.Diagnostic {
+				podTemplateSpec, found := extract.PodTemplateSpec(object.K8sObject)
+				if !found {
+					return nil
+				}
+				for _, obj := range lintCtx.Objects() {
+					networkpolicy, ok := obj.K8sObject.(*networkingV1.NetworkPolicy)
+					if !ok {
+						continue
+					}
+					if object.K8sObject.GetNamespace() != obj.K8sObject.GetNamespace() {
```

Comment:
I have a problem with this check, that makes kube-linter complain about a network policy that actually works.
This is the output of helm, and the deploy has a namespace set, but the network policy don't. Still once created, the network policy is actually working.

So do we really need that check?

Reply:
Network policies are namespaced resourced, so you do actually need it. Otherwise, if the deployment has a namespace explicitly set, but you do `helm install` passing `-n other-namespace`, then the network policy will not work. Does that make sense?

Reply:
yes it does. Thanks!